jekyll/_cci2/testing-orbs.md

```diff
@@ -0,0 +1,191 @@
+* Runtime Testing - this requires setting up separate tests and running them inside a CircleCI build.
+* Integration Testing - this is likely only needed for fairly advanced orbs or orbs designed specifically as public, stable interfaces to 3rd-party services. Doing orb integration tests requires a custom build and your own external testing environments for whatever systems you are integration with.
+
+The sections below provide details on each of these techniques for each level of orb testing.
+
+### Schema Validation
+
+To test whether an orb is valid YAML and is well-formed according to the schema, use `circleci orb validate` with the CircleCI CLI.
+
+#### Example
+
+Given an orb with source at `./src/orb.yml` you can run `circleci orb validate ./src/orb.yml` to receive feedback on whether the orb is valid and will pass through config processing. If there is an error, you will receive the first schema validation error encountered. Alternatively, you can pass STDIN rather than a file path. 
+
+For example, equivalent to the previous example you can run `cat ./src/orb.yml | circleci orb validate.` 
+
+**Note** Schema errors are often best read "inside out", where your coding error may be best described in one of the inner-most errors in a stack of errors.
+
+Validation testing may also be performed inside a build using the CircleCI CLI. You can directly use the `circleci/circleci-cli` orb to insert the CLI into your builds, or use the `circleci/orb-tools` orb that contains a few handy commands jobs, including performing schema validation by default in its jobs. An example of an orb that uses `orb-tools` to do basic publishing is the "hello-build" orb.
+
+### Expansion Testing
+
+The next level of orb testing validates that an orb is expanding to generate the desired final `config.yml` consumed by the CircleCI system.
+
+This testing is best completed by either publishing the orb as a dev version and then using it in your config and processing that config, or by using config packing to make it an inline orb and then processing that. Then you can use `circleci config` process and compare it to the expanded state you expect.
+
+Another form of expansion testing can be done on the orb itself when your orb references other orbs. You can use the `circleci orb` process to resolve the orbs-dependent orbs and see how it would be expanded when you publish it to the registry.
+
+Keep in mind that when testing expansion it's best to test the underlying structure of the data rather than the literal string that is expanded. A useful tool to make assertions about things in yaml is `yq`. This allows you to inspect specific structural elements rather than having fragile tests that depend on string comparisons or every aspect of an expanded job.
+
+The following steps illustrate doing basic expansion testing from the CLI:
+
+1) Assume a simple orb in `src/orb.yml`
+
+{% raw %}
+```
```

Comment:
yaml has to be on the same line as the ticks

Reply:
Fixed